### DISCUSSION TITLE
Building Custom Table From API Data source and interact with API

### DISCUSSION THREAD

**Discussion by devfaysal** [filamentphp/filament#8497] [General]
I want to create a custom page in Filament but do not want to use Filaments table builder.
I just want to use the Filament table's HTML and style so that the admin panel has a similar design. 
I will use my functionality for the table.
What is the best way to do that? I checked Blade components https://filamentphp.com/docs/3.x/support/blade-components/overview
But did not find anything about the table there.

My use case is below, if you have a better suggestion, please let me know.

Please have a look at the screenshot.
It's a custom admin panel I built using Vue.js.
In the screenshot, the table data is coming from a WooCommerce(WordPress e-commerce) website.
Changing the input values and clicking the update button updates the price and stock in the WooCommerce website by API.

I found someone was suggesting using [calebporzio/sushi](https://github.com/calebporzio/sushi)
But that does not serve my purpose.

![2023-09-14-17-53-46](https://github.com/filamentphp/filament/assets/16212149/f97c957c-93a7-4a4e-ba7d-67f0ca9af76e)

**Comment by leandrocfe** [comment]
hum.. I have an example but it is using sushi as well
[how-to-consume-an-external-api-with-filament-tables](https://filamentphp.com/community/how-to-consume-an-external-api-with-filament-tables)

**Reply by devfaysal** [reply]
Thank you.
I have seen your example. I mentioned your example in the post.
It's great for displaying external data from API.
But it looks complicated for me to interact with the data e.g. updating.

**Reply by leandrocfe** [reply]
You mean, updating via modal/form?

**Reply by devfaysal** [reply]
Yeah, correct.
Please have a look at the screenshot I added in the main post, I have done this using Vue.js
I am migrating the admin panel to Filament.
Want a similar table in Filament.

**Reply by leandrocfe** [reply]
You can use modals or pages to manage data creation, editing, and display through API interactions. While implementing this approach may present some challenges, such as the need to intercept and manage all actions...

**Reply by devfaysal** [reply]
Yeah, that's why I want to use my own functionality, but I want to use the Filament tables design/HTML so that it matches other parts.

For example, we can easily use Filament's button using the button component. I am looking for something similar for the table
```
<x-filament::button wire:click="openNewUserModal">
    New user
</x-filament::button>
```

**Reply by leandrocfe** [reply]
Have you ever tried this package before? https://packagist.org/packages/jgrossi/corcel

Available components here: [Filament Blade Components](https://filamentphp.com/docs/3.x/support/blade-components/overview).

However, if you find yourself needing to extensively modify Filament's default behavior, it raises the question of why you initially opted for Filament. In such a scenario, continuing to use Vue.js might be a more suitable choice.

**Reply by devfaysal** [reply]
I think I did not make everything clear in the post.
I made the app using Laravel and Vue.js a few years ago. I used WordPress API.
Everything works totally fine as expected.
The app has other features as well, it's a custom CRM.

Recently, I wanted to migrate that app to Filament because adding more features is easy.
I migrated all other features to the filament but was stuck in this only remaining feature.

**Reply by leandrocfe** [reply]
Maybe you can consider a hybrid approach by creating a custom table component to consume this API...

**Reply by devfaysal** [reply]
yeah, finally I will have to go for this maybe.

**Reply by devfaysal** [reply]
The table component is in progress.
Just got reply in Twitter from Zep
https://github.com/filamentphp/filament/issues/7105

**Reply by devfaysal** [reply]
It took some time, but finally, I was able to make the table similar and now all working fine in Filament.
![2023-09-16-14-35-21](https://github.com/filamentphp/filament/assets/16212149/96bc5738-d436-4df3-b949-880416f2e126)

**Reply by leandrocfe** [reply]
looks great!

**Comment by code-reflex** [comment]
> It took some time, but finally, I was able to make the table similar and now all working fine in Filament. ![2023-09-16-14-35-21](https://user-images.githubusercontent.com/16212149/268438011-96bc5738-d436-4df3-b949-880416f2e126.png)

Hello, Even I am trying to achieve something similar. I want a sub form OR a new form-field which can have a table like yours and the data going into a json column.  Can you please guide on how this can be achieved? how have you implemented your solution?

**Reply by devfaysal** [reply]
https://github.com/devfaysal/crm
This is the repo where I implemented my solution.

**Reply by code-reflex** [reply]
Thank you for sharing the reference repo. I will take a look. Thanks once again.
